crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -104,13 +127,18 @@ impl DWalletMPCManager {
             weighted_threshold_access_structure,
             weighted_parties,
             batched_sign_sessions: HashMap::new(),
-            outputs_manager: DWalletMPCOutputsManager::new(&epoch_store),
+            outputs_manager: outputs_manager.clone(),
```

Comment:
```suggestion
outputs_manager: DWalletMPCOutputsManager::new(&epoch_store),
```
No! This should be a new outputs manager, so when the manager handle the outputs one by one, it will catch all the malicious actors.